pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -559,26 +544,42 @@ func (rc *reconciler) updateStates(volumesNeedUpdate map[v1.UniqueVolumeName]*re
 			}
 		}
 	}
+}
+
+func getDeviceMountPath(volume *reconstructedVolume) (string, error) {
+	volumeAttacher, err := volume.attachablePlugin.NewAttacher()
+	if volumeAttacher == nil || err != nil {
+		return "", err
+	}
+	deviceMountPath, err :=
+		volumeAttacher.GetDeviceMountPath(volume.volumeSpec)
+	if err != nil {
+		return "", err
+	}
 
-	// Get the list of volumes from desired state and update OuterVolumeSpecName if the information is available
-	volumesToMount := rc.desiredStateOfWorld.GetVolumesToMount()
-	for _, volumeToMount := range volumesToMount {
-		if volume, exists := volumesNeedUpdate[volumeToMount.VolumeName]; exists {
-			volume.outerVolumeSpecName = volumeToMount.OuterVolumeSpecName
-			volumesNeedUpdate[volumeToMount.VolumeName] = volume
-			glog.V(4).Infof("Update OuterVolumeSpecName from desired state for volume (%q): %q",
-				volumeToMount.VolumeName, volume.outerVolumeSpecName)
+	if volume.blockVolumeMapper != nil {
```

Comment:
@jingxu97 
Currently blockVolumeMapper is only nil if plugin doesn't support Block volume.
If a volume plugin supports `block volume` and the volumeMode is `filesystem`, blockVolumeMapper is initialized during reconstructVolume(), so the volume.blockVolumeMapper isn't nil.
As a result, deviceMountPath return wrong information for `Filesystem` volumeMode case.

We have two choices here.
- reconstructVolume() doesn't initialize blockVolumeMapper if volumeMode is `Filesystem`.
- Check volumeMode when getDeviceMountPath() obtains deviceMountPath.

I suppose either way would work. Any thought?

Reply:
I think the first option is good for me. Update the code to fix the issue now. Thanks!